### ISSUE TITLE
[explicit-function-return-type] Rule doesn't work with class arrow methods

### ISSUE THREAD

**Issue by slikts** [typescript-eslint/typescript-eslint#348]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-function-return-type": "error"
  }
}
```

```TS
export class Foo {
  bar() { return 123; } // Error
  baz = () => 123; // Missing error
}
export const foo = () => 123; // Error
```

**Expected Result**

Class arrow method with no return type annotation should be an error.

**Actual Result**

No error.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.2` |
| `@typescript-eslint/parser`        | `1.4.2` |
| `TypeScript`                       | `3.3.3` |
| `ESLint`                           | `5.14.1` |
